dataland-frontend/src/components/forms/CreateEuTaxonomyForFinancials.vue

```diff
@@ -0,0 +1,655 @@
+          for (let index = 0; index < this.filesToUpload.length; index++) {
+            const uploadFileSuccessful = await documentUploadControllerControllerApi.postDocument(
+              this.filesToUpload[index]
+            );
+            if (!uploadFileSuccessful) {
+              allFileUploadedSuccessful = false;
+              break;
+            } else if (uploadFileSuccessful) {
+              this.filesToUpload = [
+                ...this.updatePropertyFilesUploaded(
+                  index,
+                  "documentId",
+                  uploadFileSuccessful.data.documentId,
+                  this.filesToUpload
+                ),
+              ];
+            }
+          }
+        }
+        if (allFileUploadedSuccessful) {
+          await this.$nextTick();
+          const formInputsModelToSend = modifyObjectKeys(
+            JSON.parse(JSON.stringify(this.formInputsModel)) as ObjectType,
+            "send"
+          );
+          this.postEuTaxonomyDataForFinancialsResponse =
+            await euTaxonomyDataForFinancialsControllerApi.postCompanyAssociatedEuTaxonomyDataForFinancials(
+              formInputsModelToSend
+            );
+        }
+      } catch (error) {
+        console.error(error);
+      } finally {
+        this.postEuTaxonomyDataForFinancialsProcessed = true;
+        this.confirmedSelectedKPIs = [];
+        this.selectedKPIs = [];
+        this.fiscalYearEndAsDate = null;
+        this.filesToUpload = [];
+        this.listOfUploadedReportsInfo = [];
+        this.formInputsModel = {};
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        this.$refs.UploadReports.clearAllSelectedFiles();
+        await this.$nextTick();
+        this.$formkit.reset("CreateEuTaxonomyForFinancialsForm");
+      }
+    },
+
+    /**
+     * Add files to object filesToUpload
+     *
+     * @param event full event object containing the files
+     * @param event.originalEvent event information
+     * @param event.files files
+     */
+    onSelectedFilesHandler(event: { files: Record<string, string>[]; originalEvent: Event }): void {
+      if (event.files.length) {
+        this.filesToUpload = [
+          ...completeInformationAboutSelectedFileWithAdditionalFields(event.files, this.listOfUploadedReportsInfo),
+        ] as ExtendedFile[];
+      } else {
+        return;
+      }
+    },
+
+    /**
+     * Remove report from files uploaded
+     *
+     * @param fileToRemove File To Remove
+     * @param fileRemoveCallback Callback function removes report from the ones selected in formKit
+     * @param index Index number of the report
+     */
+    removeReportFromFilesToUpload(
+      fileToRemove: Record<string, string>,
+      fileRemoveCallback: (x: number) => void,
+      index: number
+    ) {
+      fileRemoveCallback(index);
+      this.filesToUpload = this.filesToUpload.filter((el) => {
+        return el.name !== fileToRemove.name;
+      });
+    },
```

Comment:
Why does the fileToRemove need to be passed to this methods as param?
You already have the `index` of the file which needs to be removed.
Why don't you just do:
      `this.filesToUpload.splice(index, 1)` ?

Reply:
We can do that and it will work, but I think it's better to return a new array than to modify the one that is directly.

Reply:
Reminder: Discuss this with other team members

Reply:
I did this centrally for the UploadReports.vue component. So we can close this conversation.